mozregression/telemetry.py

```diff
@@ -0,0 +1,55 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import sys
+from multiprocessing import Process
+
+from mozlog import get_proxy_logger
+
+
+LOG = get_proxy_logger('telemetry')
+
+
+def _send_telemetry_ping(variant, appname):
+    from pathlib import Path
+    from glean import Glean, Configuration
+    from glean import (load_metrics,
+                       load_pings)
+    from mozregression import __version__
+
+    mozregression_path = Path.home() / '.mozilla' / 'mozregression'
+    Glean.initialize(
+        application_id="mozregression",
+        application_version=__version__,
+        upload_enabled=True,
+        configuration=Configuration(
+            ping_tag="mozregression-test-tag"
+        ),
+        data_dir=mozregression_path / "data"
+    )
+
+    # if we got this far, we can assume telemetry is enabled
+    Glean.set_upload_enabled(True)
+
+    pings = load_pings("pings.yaml")
+    metrics = load_metrics("metrics.yaml")
+
+    metrics.usage.variant.set(variant)
+    metrics.usage.app.set(appname)
+
+    pings.usage.submit()
+
+
+def send_telemetry_ping(variant, appname):
+    if sys.version_info[0] < 3:
+        # sending telemetry pings is python3-only, since
+        # it relies on Glean
+        LOG.info("Not using python 3+, not sending telemetry ping")
+        return
+
+    LOG.info("Sending anonymized telemetry recording mozregression "
+             "usage, learn more at: https://mozilla.github.io/mozregression/telemetry.html")
+
+    p = Process(target=_send_telemetry_ping, args=(variant, appname))
+    p.start()
```

Comment:
This totally makes sense to not block on I/O.  This might change as part of [a bug to provide built-in multiprocessing for Glean](https://bugzilla.mozilla.org/show_bug.cgi?id=1604191), but we can come back around to this then.